pkg/storegateway/batch_series_test.go

```diff
@@ -771,52 +773,169 @@ func TestBlockSeriesChunkRefsSetsIterator_ErrorPropagation(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	t.Cleanup(cancel)
 
-	suite := prepareStoreWithTestBlocks(t, t.TempDir(), objstore.NewInMemBucket(), false, NewChunksLimiterFactory(0), NewSeriesLimiterFactory(0))
-	var firstBlock *bucketBlock
-	for _, b := range suite.store.blocks {
-		firstBlock = b
-		break
+	testCases := map[string]struct {
+		matcher        *labels.Matcher
+		batchSize      int
+		chunksLimit    int
+		seriesLimit    int
+		expectedErr    string
+		expectedSeries []seriesChunkRefsSet
+	}{
+		"chunks limits reached": {
+			matcher:     labels.MustNewMatcher(labels.MatchRegexp, "a", ".+"),
+			batchSize:   100,
+			chunksLimit: 1,
+			seriesLimit: 100,
+			expectedErr: "test limit exceeded",
+		},
+		"series limits reached": {
+			matcher:     labels.MustNewMatcher(labels.MatchRegexp, "a", ".+"),
+			batchSize:   100,
+			chunksLimit: 100,
+			seriesLimit: 1,
+			expectedErr: "test limit exceeded",
+		},
+		"selects all series": {
+			matcher:     labels.MustNewMatcher(labels.MatchRegexp, "a", ".+"),
+			batchSize:   100,
+			chunksLimit: 100,
+			seriesLimit: 100,
+			expectedSeries: []seriesChunkRefsSet{
+				{series: []seriesChunkRefs{
+					{lset: labels.FromStrings("a", "1", "b", "1"), chunks: make([]seriesChunkRef, 1)},
```

Comment:
I would remove the chunks cause doesn't add much value. Then I would change the assertions (see comment below).

```suggestion
					{lset: labels.FromStrings("a", "1", "b", "1")},
```